src/Hakyll/Core/Compiler/Internal.hs

```diff
@@ -164,25 +165,28 @@ instance MonadMetadata Compiler where
 --------------------------------------------------------------------------------
 instance MonadError [String] Compiler where
   throwError = compilerThrow
-  catchError = compilerCatch
+  catchError = (. matchErr) . compilerCatch
+    where
+      matchErr f Logger.Error es = f es
+      matchErr f _            _  = f []
```

Comment:
I do wonder whether this is a good idea. If a compiler failed with some verbosity below `Error`, why give no information to the handler at all? It might be useful.
Still, the argument is that ```compile `catchError` throwError``` should behave more like identity, and not change the verbosity level.